---
title: "R Notebook"
output: html_notebook
---

#Heterozygosity of the old dataset

## Heterozygosity of the full dataset:

```{r}

library(dplyr)
library(magrittr)
library(tidyr)

het <- read.delim(file = "/run/user/1000/gvfs/sftp:host=cbsugore02.tc.cornell.edu/workdir/rjl278/Sorghum/Imputation/het.table", header = T) %>% 
  rename(HOM1=OBS.HOM1, HOM2=HOM2. , ID=CHR_POS) %>%
  mutate(P.Het =  HET/(HOM1+HET+HOM2))

hist(het$P.Het, xlab= "% Heterozygosity", main="")


```

```{r}
library(scales)

maf_table <- het %>%
  sample_frac(.1) %>%
  mutate(MAF = case_when( HOM1 > HOM2 ~ (HET+HOM2*2)/(2*(HET+HOM1+HOM2)), 
                          HOM1 < HOM2 ~ (HET+HOM1*2)/(2*(HET+HOM1+HOM2)))) 


plot(maf_table$MAF, maf_table$P.Het, pch=1, cex=0.4, col=alpha("black",alpha = 0.1), xlab = "MAF", ylab = "% heterozygosity" )


```

```{r, fig.width= 15, fig.height=5}

hetperchr <- het %>%
  sample_frac(.2) %>%
  select(ID, P.Het) %>%
  mutate(tempcol = stringr::str_split(ID, '_')) %>%
  rowwise() %>%
  mutate(source = unlist(tempcol)[1], medium = unlist(tempcol)[2])%>%
  select(-tempcol) %>%
  set_colnames(c("SNP","n","chr", "BP")) %>%
  mutate(CHR = substr(chr, 4,7)) %>%
  select(-chr)


hetperchr$CHR <- as.integer(as.numeric(as.character(hetperchr$CHR)))
hetperchr$BP <- as.integer(as.numeric(as.character(hetperchr$BP)))
head(hetperchr)

library(qqman)
library(wesanderson)
wes <- wes_palette("Cavalcanti")

#manhattan(subset(snpden, CHR ==5, cex = 0.5), p = "n", logp = FALSE, ylab = "SNPs/kb", genomewideline = FALSE, col = wes,
#    suggestiveline = FALSE, main = "SNP density", ylim = (c(0,50)) )

manhattan(hetperchr, cex = 0.1, p = "n", logp = FALSE, ylab = "%het", genomewideline = FALSE, col = alpha(wes,alpha=0.1),
    suggestiveline = FALSE, main = "%het chromosome distribution", ylim = (c(0,0.15)) )

```



## Heterozygosity in deleterious positions

```{r}

deleterious <- read.delim(file = "/run/user/1000/gvfs/sftp:host=cbsugore02.tc.cornell.edu/workdir/rjl278/Sorghum/Imputation/Deleterious.list", header = F)

del_table <- het %>%
  filter(ID %in% deleterious[,1])

hist(del_table$P.Het, xlab= "% Heterozygosity", main = "")

```

# Heterozygosity by MAF 

```{r}
library(scales)

maf_table <- het %>%
  filter(ID %in% deleterious[,1]) %>%
  mutate(MAF = case_when( HOM1 > HOM2 ~ (HET+HOM2*2)/(2*(HET+HOM1+HOM2)), 
                          HOM1 < HOM2 ~ (HET+HOM1*2)/(2*(HET+HOM1+HOM2)))) 


plot(maf_table$MAF, maf_table$P.Het, pch=1, cex=0.4, col=alpha("black",alpha = 0.2), xlab = "MAF", ylab = "% heterozygosity" )


```

```{r, fig.width= 15, fig.height=5}


hetperchr <- del_table %>%
  select(ID, P.Het) %>%
  mutate(tempcol = stringr::str_split(ID, '_')) %>%
  rowwise() %>%
  mutate(source = unlist(tempcol)[1], medium = unlist(tempcol)[2])%>%
  select(-tempcol) %>%
  set_colnames(c("SNP","n","chr", "BP")) %>%
  mutate(CHR = substr(chr, 4,7)) %>%
  select(-chr)


hetperchr$CHR <- as.integer(as.numeric(as.character(hetperchr$CHR)))
hetperchr$BP <- as.integer(as.numeric(as.character(hetperchr$BP)))
head(hetperchr)

library(qqman)
library(wesanderson)
wes <- wes_palette("Cavalcanti")

#manhattan(subset(snpden, CHR ==5, cex = 0.5), p = "n", logp = FALSE, ylab = "SNPs/kb", genomewideline = FALSE, col = wes,
#    suggestiveline = FALSE, main = "SNP density", ylim = (c(0,50)) )

manhattan(hetperchr, cex = 0.5, p = "n", logp = FALSE, ylab = "%het", genomewideline = FALSE, col = wes,
    suggestiveline = FALSE, main = "Deleterious alleles %het chromosome distribution", ylim = (c(0,0.15)) )



```





```{r}

M5 <- read.delim(file = "/run/user/1000/gvfs/sftp:host=cbsugore02.tc.cornell.edu/workdir/rjl278/Sorghum/Imputation/5M.het", header = T)

hist(M5$F, breaks = 20)
plot(density(M5$F), ylab = "number of taxa", xlab= "Inbreeding coefficient", main="" )
```

```{r}

M5_imp <- read.delim(file = "/run/user/1000/gvfs/sftp:host=cbsugore02.tc.cornell.edu/workdir/rjl278/Sorghum/Imputation/5M_imputed.het", header = T)

hist(M5_imp$F, breaks = 20)
plot(density(M5_imp$F), ylab = "number of taxa", xlab= "Inbreeding coefficient", main="" )
```




```{r}

M5_new <- read.delim(file = "/run/user/1000/gvfs/sftp:host=cbsugore02.tc.cornell.edu/workdir/rjl278/Sorghum/Imputation/TERRA_435_5M.het", header = T)

hist(M5_new$F, breaks = 20, probability = T)
#plot(density(M5_new$F), ylab = "number of taxa", xlab= "Inbreeding coefficient", main="", xlim=c(0,1) )
lines(density(M5_new$F, adjust = 1), col="blue", lwd=1)

```

```{r}

Ftable <- read.delim(file = "/run/user/1000/gvfs/sftp:host=cbsugore02.tc.cornell.edu/workdir/rjl278/Sorghum/WGS/PROBLEM/table.f", header = T)

library(ggplot2)

n <- ggplot(Ftable, aes(y=NEWF,x=Batch)) +
  geom_boxplot() +
  ylim(0,1)
n

```


```{r}

p <- ggplot(Ftable, aes(y=EF,x=Batch)) +
  geom_boxplot()+
  ylim(0,1)
p


```


```{r}

p <- ggplot(Ftable, aes(y=EF,x=Batch)) +
  geom_boxplot()+
  ylim(0,1)
p

ggplot(Ftable, aes(x=EF, y=NEWF, color=Batch)) + 
  geom_point()+
  geom_smooth(method=lm, color="black")+
  labs(x="Elodie", y = "Roberto")+
  scale_color_brewer(palette="Paired") + theme_classic() + 
  geom_abline(intercept = 0)




```


Heterozigosity by depth

```{r}

Ftable <- read.delim(file = "/run/user/1000/gvfs/sftp:host=cbsugore02.tc.cornell.edu/workdir/rjl278/Sorghum/WGS/PROBLEM/table.f", header = T)
Dtable <- read.delim(file = "/run/user/1000/gvfs/sftp:host=cbsugore02.tc.cornell.edu/workdir/rjl278/Sorghum/WGS/PROBLEM/table.depth", header = F)
colnames(Dtable) <- c("WGS_Name", "depth")
Dtable

corre <- inner_join(Ftable, Dtable, by="WGS_Name")


ggplot(corre, aes(x=depth, y=EF, color=Batch)) + 
  geom_point()+
  #geom_smooth(method=lm, color="black")+
  labs(x="Depth", y = "Inbreeding coefficient")+
  scale_color_brewer(palette="Paired") + theme_classic()



```

```{r}

ggplot(corre, aes(x=EF, y=Ereads, color=Batch)) + 
  geom_point()+
  #geom_smooth(method=lm, color="black")+
  labs(x="Depth", y = "Number of genotypes called")+
  scale_color_brewer(palette="Paired") + theme_classic()


```


Add a new chunk by clicking the *Insert Chunk* button on the toolbar or by pressing *Ctrl+Alt+I*.

When you save the notebook, an HTML file containing the code and output will be saved alongside it (click the *Preview* button or press *Ctrl+Shift+K* to preview the HTML file).

The preview shows you a rendered HTML copy of the contents of the editor. Consequently, unlike *Knit*, *Preview* does not run any R code chunks. Instead, the output of the chunk when it was last run in the editor is displayed.
