---
title: "R Notebook"
output: html_notebook
---

# Derived Allele Frequencies (DAF)

```{r, fig.width=10, fig.height=4}


library(dplyr)
library(magrittr)
library(tidyr)
library(ggplot2)



daf <- read.delim( file = "/home/roberto/Sorghum/Evol_model/Derived_alleles/splitting/FULL.DAF", header =  T) %>%
  mutate(Class = case_when(DAF_p >= 0.55 & Ref_count > Alt_count ~ "Ancestral",
                          DAF_p >= 0.55 & Alt_count > Ref_count ~ "Derived",
                          DAF_p <=0.45 & Ref_count > Alt_count ~ "Derived",
                          DAF_p <=0.45 & Alt_count > Ref_count ~ "Ancestral",
                          DAF_p <0.55 & DAF_p > 0.45 ~ "NA") ) %>%
  mutate(DAF = case_when(Class == "Ancestral" ~ MAF,
                         Class == "Derived" ~ 1-MAF) ) %>%
  dplyr::select(DAF, SET) %>%
  na.omit(DAF)


#######  Deleterious.-

del <- daf
del$bins <- cut(del$DAF, breaks=c(0,0.05,0.1,0.15,0.2,0.25,0.3,0.35,0.4,0.45,0.5,0.55,0.6,0.65,0.7,0.75,0.8,0.85,0.9,0.95,1), 
                labels=c("1","2","3", "4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20")) 
del <- del %>%
  filter(SET == "Deleterious")

de2m            <- as.data.frame(matrix(nrow = 20, ncol = 3,0))
colnames(de2m)  <- c("SET", "Proportion", "DAF_group")
de2m$SET        <- "Deleterious"
de2m$Proportion <- prop.table(table(del$bins))
de2m$DAF_group  <- c("1","2","3", "4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20")

#######  Deleterios_nc.-

delnc <- daf
delnc$bins <- cut(delnc$DAF, breaks=c(0,0.05,0.1,0.15,0.2,0.25,0.3,0.35,0.4,0.45,0.5,0.55,0.6,0.65,0.7,0.75,0.8,0.85,0.9,0.95,1), 
                labels=c("1","2","3", "4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20")) 
delnc <- delnc %>%
  filter(SET == "Deleterious_nc")

denc2m            <- as.data.frame(matrix(nrow = 20, ncol = 3,0))
colnames(denc2m)  <- c("SET", "Proportion", "DAF_group")
denc2m$SET        <- "Deleterious_nc"
denc2m$Proportion <- prop.table(table(delnc$bins))
denc2m$DAF_group  <- c("1","2","3", "4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20")

#######  STOP .-

sto <- daf
sto$bins <- cut(sto$DAF, breaks=c(0,0.05,0.1,0.15,0.2,0.25,0.3,0.35,0.4,0.45,0.5,0.55,0.6,0.65,0.7,0.75,0.8,0.85,0.9,0.95,1), 
                labels=c("1","2","3", "4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20")) 
sto <- sto %>%
  filter(SET == "Stops")

sto2m            <- as.data.frame(matrix(nrow = 20, ncol = 3,0))
colnames(sto2m)  <- c("SET", "Proportion", "DAF_group")
sto2m$SET        <- "Stops"
sto2m$Proportion <- prop.table(table(sto$bins))
sto2m$DAF_group  <- c("1","2","3", "4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20")


#######  Tolerant.-

tol <- daf
tol$bins <- cut(tol$DAF, breaks=c(0,0.05,0.1,0.15,0.2,0.25,0.3,0.35,0.4,0.45,0.5,0.55,0.6,0.65,0.7,0.75,0.8,0.85,0.9,0.95,1), 
                labels=c("1","2","3", "4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20")) 
tol <- tol %>%
  filter(SET == "Tolerated")

tol2m            <- as.data.frame(matrix(nrow = 20, ncol = 3,0))
colnames(tol2m)  <- c("SET", "Proportion", "DAF_group")
tol2m$SET        <- "Tolerated"
tol2m$Proportion <- prop.table(table(tol$bins))
tol2m$DAF_group  <- c("1","2","3", "4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20")

#######  Synonymous.-

syn <- daf
syn$bins <- cut(syn$DAF, breaks=c(0,0.05,0.1,0.15,0.2,0.25,0.3,0.35,0.4,0.45,0.5,0.55,0.6,0.65,0.7,0.75,0.8,0.85,0.9,0.95,1), 
                labels=c("1","2","3", "4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20")) 
syn <- syn %>%
  filter(SET == "Synonymous")

syn2m            <- as.data.frame(matrix(nrow = 20, ncol = 3,0))
colnames(syn2m)  <- c("SET", "Proportion", "DAF_group")
syn2m$SET        <- "Synonymous"
syn2m$Proportion <- prop.table(table(syn$bins))
syn2m$DAF_group  <- c("1","2","3", "4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20")


## MERGE THE DATA:

merged <- rbind(de2m, denc2m, sto2m, tol2m, syn2m)
merged <- merged%>%
  mutate(YYY = case_when (DAF_group == "1" ~ "0.00-0.05",
                          DAF_group == "2" ~ "0.05-0.10",
                          DAF_group == "3" ~ "0.10-0.15",
                          DAF_group == "4" ~ "0.15-0.20",
                          DAF_group == "5" ~ "0.20-0.25",
                          DAF_group == "6" ~ "0.25-0.30",
                          DAF_group == "7" ~ "0.30-0.35",
                          DAF_group == "8" ~ "0.35-0.40",
                          DAF_group == "9" ~ "0.40-0.45",
                          DAF_group == "10" ~ "0.45-0.50",
                          DAF_group == "11" ~ "0.50-0.55",
                          DAF_group == "12" ~ "0.55-0.60",
                          DAF_group == "13" ~ "0.60-0.65",
                          DAF_group == "14" ~ "0.65-0.70",
                          DAF_group == "15" ~ "0.70-0.75",
                          DAF_group == "16" ~ "0.75-0.80",
                          DAF_group == "17" ~ "0.80-0.85",
                          DAF_group == "18" ~ "0.85-0.90",
                          DAF_group == "19" ~ "0.90-0.95",
                          DAF_group == "20" ~ "0.95-1.00") )



merged$DAF_group <- as.factor(merged$DAF_group)
levels(merged$DAF_group)
merged$DAF_group <- factor(merged$DAF_group, levels = c("1","2","3", "4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20"))

## PLOT

library(wesanderson)
library(ggplot2)
wes =wes_palette("Cavalcanti1")
weso = c(wes[1],wes[2],wes[5],wes[3],wes[4])


Graph <- ggplot(data=merged, aes(x=YYY, y=Proportion, fill=SET)) +
  geom_bar(stat="identity", position=position_dodge())+
  ylab("Proportion of SNPs") +
  xlab("Derived allele frequencies (DAF)") +
  scale_fill_manual(values=weso)+
  theme_minimal() +
  theme(axis.text.x = element_text(face="bold", size=9, angle=90)) +
  guides(fill=guide_legend(title=""))

Graph 

#ggsave("/home/roberto/Sorghum/Figures/Fig2.png", plot= Graph, device = "png", scale = 1, width = 9, height = 4, units = c("in"),dpi = 550, limitsize = TRUE)


```

#Synteny SORGHUM

```{r, fig.height=5, fig.width=4}

Synteny <- read.delim( file = "/home/roberto/Sorghum/Evol_model/Derived_alleles/splitting/FULL.DAF", header =  T, stringsAsFactors = FALSE) %>%
  dplyr::select(SET, Group)


table(Synteny$Group)

Non_fractioned <- rename(count(Synteny, SET, Group), Freq = n) %>% 
  filter(Group == "Non_fractioned") %>%
  mutate(Percentage = Freq/sum(Freq))
Fractioned <- rename(count(Synteny, SET, Group), Freq = n) %>% 
  filter(Group == "Fractioned") %>%
  mutate(Percentage = Freq/sum(Freq))
NonSyntenic <- rename(count(Synteny, SET, Group), Freq = n) %>% 
  filter(Group == "Sorghum_only") %>%
  mutate(Percentage = Freq/sum(Freq))
  
Merged <- rbind(Non_fractioned, Fractioned, NonSyntenic) %>%
  mutate(Group = replace(Group, Group == "Sorghum_only", "NonSyntenic")) %>%
  mutate(Group = replace(Group, Group == "Non_fractioned", "NonFractioned"))


wes =wes_palette("Cavalcanti1")
weso = c(wes[1],wes[2],wes[5],wes[3],wes[4])

Sorghum_del <- ggplot() + theme_bw() +
  geom_bar(aes(y = Percentage, x = Group, fill = SET), data = Merged, stat="identity", width = 0.7) +
  theme(legend.position="none", legend.direction="horizontal",
        legend.title = element_blank()) +
  scale_fill_manual(values=weso) +
  xlab("") + 
  ylab("Proportion")
  
Sorghum_del

```

#Synteny MAIZE

```{r, fig.height=5, fig.width=4}

GERP <- read.delim(file = "/home/roberto/Maize/Comparative_Hapmap/DEL/GERP.ready", header = F, stringsAsFactors = FALSE) %>%
  `colnames<-`(c("link", "GERP"))
SIFT <- read.delim(file = "/home/roberto/Maize/Comparative_Hapmap/DEL/SIFT_unix.noalternates", stringsAsFactors = FALSE)%>%
  rename(link = CHROM_POS) %>%
  left_join(GERP, by="link")


#########################################################################################

deleterious <- SIFT %>%
  filter(as.numeric(as.character(SIFT_SCORE)) <= 0.05 & GERP >= 2) %>%
  mutate(CLASS = "deleterious")

deleterious_nc <- SIFT %>%
  filter(as.numeric(as.character(SIFT_SCORE)) <= 0.05 & GERP < 2) %>%
  mutate(CLASS = "deleterious_nc")

stops <- SIFT %>%
  filter(VARIANT_TYPE == "STOP-GAIN" | VARIANT_TYPE == "STOP-LOSS") %>%
  mutate(CLASS = "stops")

tolerated <- SIFT %>%
  filter(VARIANT_TYPE == "NONSYNONYMOUS" & SIFT_PREDICTION =="TOLERATED" ) %>%
  mutate(CLASS = "tolerated")

synonymous <- SIFT %>%
  filter(VARIANT_TYPE == "SYNONYMOUS" ) %>%
  mutate(CLASS = "synonymous")



#########################################################################################  

fracs <- read.delim("/home/roberto/Maize/Comparative_Hapmap/DEL/Maize.fractionation", header = T, stringsAsFactors = FALSE)

joined <- rbind(deleterious, deleterious_nc, stops, tolerated, synonymous) %>%
  left_join(fracs, by="TRANSCRIPT_ID")


table(joined$CLASS)
#save(joined, file= "/home/roberto/docker/Maize.joined_2.Rdata")
#load("/home/roberto/docker/Maize.joined.Rdata")


Panaliza <- joined %>%
  dplyr::select(CLASS, FRAC) %>%
  rename(SET=CLASS) %>%
  rename(Group=FRAC)

Non_fractioned <- rename(count(Panaliza, SET, Group), Freq = n) %>% 
  filter(Group == "NonFractioned") %>%
  mutate(Percentage = Freq/sum(Freq))
Fractioned <- rename(count(Panaliza, SET, Group), Freq = n) %>% 
  filter(Group == "Fractioned") %>%
  mutate(Percentage = Freq/sum(Freq))
NonSyntenic <- rename(count(Panaliza, SET, Group), Freq = n) %>% 
  filter(Group == "NS") %>%
  mutate(Percentage = Freq/sum(Freq))
  
Merged <- rbind(Non_fractioned, Fractioned, NonSyntenic) %>%
  mutate(Group = replace(Group, Group == "NS", "NonSyntenic"))


wes =wes_palette("Cavalcanti1")
weso = c(wes[1],wes[2],wes[5],wes[3],wes[4])

Maize_del <- ggplot() + theme_bw() +
  geom_bar(aes(y = Percentage, x = Group, fill = SET), data = Merged, stat="identity", width = 0.7) +
  theme(legend.position="none", legend.direction="horizontal",
        legend.title = element_blank()) +
  scale_fill_manual(values=weso) +
  xlab("") + 
  ylab("")
  
Maize_del



```

#Joining Maize and Sorghum

```{r, fig.width= 9, fig.height=6}

Merged <- Graph + (Sorghum_del + Maize_del) + plot_layout(ncol = 1)  

#ggsave("/home/roberto/Sorghum/Figures/Fig2_full.png", plot= Merged, device = "png", scale = 1, width = 9, height = 5.5, units = c("in"),dpi = 750, limitsize = TRUE)

Merged


```

# Getting some stats


```{r}

Synteny <- read.delim( file = "/home/roberto/Sorghum/Evol_model/Derived_alleles/splitting/FULL.DAF", header =  T, stringsAsFactors = FALSE) 

NS <- Synteny %>%
  filter(Type=="NONSYNONYMOUS")  %>%
  filter(SIFT_score < 0.05)

```

#BURDEN


```{r, fig.width=10, fig.height=4}

library(dplyr)
library(magrittr)
library(tidyr)
library(ggplot2)


deleterious <- read.delim( file = "/home/roberto/Sorghum/Evol_model/Derived_alleles/splitting/FULL.DAF", header =  T) %>%
  mutate(Class = case_when(DAF_p >= 0.65 & Ref_count > Alt_count ~ "Ancestral",
                          DAF_p >= 0.65 & Alt_count > Ref_count ~ "Derived",
                          DAF_p <=0.35 & Ref_count > Alt_count ~ "Derived",
                          DAF_p <=0.35 & Alt_count > Ref_count ~ "Ancestral",
                          DAF_p <0.65 & DAF_p > 0.35 ~ "NA") ) %>%
  mutate(DAF = case_when(Class == "Ancestral" ~ MAF,
                         Class == "Derived" ~ 1-MAF) ) %>%
  dplyr::select(Chr, Position, SET, Class, MAF) %>%
  filter(SET != "Synonymous") %>%
  filter(SET != "Tolerated") %>%
  na.omit(Class)


del <- deleterious %>%
  filter(Class != "NA")

write.table(x = del, file = "/home/roberto/Sorghum/Evol_model/Burden/burden_manuscript.table", sep = "\t", row.names = FALSE, quote = F )

```

## Merging del info with dosage file 

```{r}

info <- read.delim("/home/roberto/Sorghum/Evol_model/Burden/burden_manuscript_ready.table") %>%
  select(CHR_POS, SET, Class, MAF)

markers <- read.delim("/home/roberto/Sorghum/Evol_model/Burden/Dosage.superready") %>%
  select(CHR_POS, COUNTED, ALT) %>%
  inner_join(info, by="CHR_POS")

header <- read.delim("/home/roberto/Sorghum/Evol_model/Burden/header", header = F) 

matriz <- read.delim("/home/roberto/Sorghum/Evol_model/Burden/Dosage.superready") %>%
  select(-COUNTED, -ALT) %>%
  `colnames<-`(header[,1])
  
Master <- markers %>%
  inner_join(matriz, by="CHR_POS")

save(Master, file = "/home/roberto/Sorghum/Evol_model/Burden/Burden_master_manuscript.Rdata")
load("/home/roberto/Sorghum/Evol_model/Burden/Burden_master_manuscript.Rdata")

```




## Deleterious alleles as minor alleles

```{r}

#Calculating different burden levels
library(ggplot2)

header <- read.delim("/home/roberto/Sorghum/Evol_model/Burden/header") 

burden_table <- as.data.frame(matrix(nrow= 485,ncol=2))
burden_table[,1] <- header
colnames(burden_table) <- c("ID", "Burden")


valores <- vector()
for (i in 7:491){
  a <- table(Master[,i])
  b <- a["2"]/sum(a)
  valores <- c(valores, b)
}

burden_table$Burden <- valores

race <- read.delim("/home/roberto/Sorghum/Evol_model/Burden/race", header = F) %>%
  `colnames<-`(c("ID", "Race"))

only_race <- race %>%
  inner_join(burden_table, by="ID")


pal <- wes_palette("FantasticFox1", 8, type = "continuous")
wes <- wes_palette("Cavalcanti1")
chev <- wes_palette("Chevalier1")
fox <- wes_palette("FantasticFox1")

colors <- c( fox[1], fox[3], "grey", wes[1], chev[1],"red", "gray31", fox[5])

p<-ggplot(only_race, aes(x=Race, y=Burden, fill=Race)) +
  geom_boxplot() +
  scale_fill_manual(values = colors)+
  theme(axis.line = element_line(size=1, colour = "black"),
        panel.grid.major = element_line(colour = "#d3d3d3"), panel.grid.minor = element_blank(),
        panel.border = element_blank(), panel.background = element_blank()) +
  #scale_y_continuous(labels = dollar_format(suffix = "%", prefix = "")) +
  theme(plot.title = element_text(size = 14, face = "bold"),
        axis.text.x=element_text(colour="black", size = 10),
        axis.text.y=element_text(colour="black", size = 10),
        legend.key = element_rect(fill = "transparent", colour = "transparent")) +
  ylim(0,0.13)
p




```

## Modifying MAF < 0.05

```{r}

#Calculating different burden levels
library(ggplot2)
library(wesanderson)
library(scales)

header <- read.delim("/home/roberto/Sorghum/Evol_model/Burden/header") 

burden_table <- as.data.frame(matrix(nrow= 485,ncol=2))
burden_table[,1] <- header
colnames(burden_table) <- c("ID", "Burden")

littlemaster <- Master %>%
  filter(MAF >0.03)

valores <- vector()
for (i in 7:491){
  a <- table(littlemaster[,i])
  b <- a["2"]/sum(a)
  valores <- c(valores, b)
}

burden_table$Burden <- valores

race <- read.delim("/home/roberto/Sorghum/Evol_model/Burden/race", header = F) %>%
  `colnames<-`(c("ID", "Race"))

only_race <- race %>%
  inner_join(burden_table, by="ID")


pal <- wes_palette("FantasticFox1", 8, type = "continuous")
wes <- wes_palette("Cavalcanti1")
chev <- wes_palette("Chevalier1")
fox <- wes_palette("FantasticFox1")

colors <- c( fox[1], fox[3], "grey", wes[1], chev[1],"red", "gray31", fox[5])


p<-ggplot(only_race, aes(x=Race, y=Burden, fill=Race)) +
  geom_boxplot() +
  scale_fill_manual(values = colors)+
  theme(axis.line = element_line(size=1, colour = "black"),
        panel.grid.major = element_line(colour = "#d3d3d3"), panel.grid.minor = element_blank(),
        panel.border = element_blank(), panel.background = element_blank()) +
  #scale_y_continuous(labels = dollar_format(suffix = "%", prefix = "")) +
  theme(plot.title = element_text(size = 14, face = "bold"),
        axis.text.x=element_text(colour="black", size = 10),
        axis.text.y=element_text(colour="black", size = 10),
        legend.key = element_rect(fill = "transparent", colour = "transparent")) 
  #ylim(0,0.1)
p

```

## Only deleterious with GERP > 2

```{r,fig.width=8, fig.height=3}

#Calculating different burden levels

Ready_del <- Master %>%
  filter(SET == "Deleterious")

header <- read.delim("/home/roberto/Sorghum/Evol_model/Burden/header") 

burden_table <- as.data.frame(matrix(nrow= 485,ncol=2))
burden_table[,1] <- header
colnames(burden_table) <- c("ID", "Burden")


valores <- vector()
for (i in 7:491){
  a <- table(Ready_del[,i])
  b <- a["2"]/sum(a)
  valores <- c(valores, b)
}


burden_table$Burden <- valores

race <- read.delim("/home/roberto/Sorghum/Evol_model/Burden/race", header = F) %>%
  `colnames<-`(c("ID", "Race"))

only_race <- race %>%
  inner_join(burden_table, by="ID")

pal <- wes_palette("FantasticFox1", 8, type = "continuous")
wes <- wes_palette("Cavalcanti1")
chev <- wes_palette("Chevalier1")
fox <- wes_palette("FantasticFox1")

colors <- c( fox[1], fox[3], "grey", wes[1], chev[1],"red", "gray31", fox[5])

p<-ggplot(only_race, aes(x=Race, y=Burden, fill=Race)) +
  geom_boxplot() +
  scale_fill_manual(values = colors)+
  theme(axis.line = element_line(size=1, colour = "black"),
        panel.grid.major = element_line(colour = "#d3d3d3"), panel.grid.minor = element_blank(),
        panel.border = element_blank(), panel.background = element_blank()) +
  #scale_y_continuous(labels = dollar_format(suffix = "%", prefix = "")) +
  theme(plot.title = element_text(size = 14, face = "bold"),
        axis.text.x=element_text(colour="black", size = 10),
        axis.text.y=element_text(colour="black", size = 10),
        legend.key = element_rect(fill = "transparent", colour = "transparent")) 
p


```


# BURDEN - Derived

```{r}

load("/home/roberto/Sorghum/Evol_model/Burden/Burden_master_manuscript.Rdata")

derivao <- Master %>%
  filter(Class == "Derived")

for (i in 7:491){
  derivao[,i][derivao[,i] == 0] <- "ZERO"
  derivao[,i][derivao[,i] == 2] <- "TWO"
}

for (i in 7:491){
  derivao[,i][derivao[,i] == "ZERO"] <- as.numeric(2)
  derivao[,i][derivao[,i] == "TWO"] <- as.numeric(0)
  derivao[,i] <- as.numeric(derivao[,i])
}
  

##
ancestral <- Master %>%
  filter(Class == "Ancestral") 


Ready <- rbind(ancestral, derivao)


table(Master$SET)
```

## All deleterious alleles

```{r, fig.height=4, fig.width=9}

#Calculating different burden levels
library(ggplot2)

header <- read.delim("/home/roberto/Sorghum/Evol_model/Burden/header") 

burden_table <- as.data.frame(matrix(nrow= 485,ncol=2))
burden_table[,1] <- header
colnames(burden_table) <- c("ID", "Burden")


valores <- vector()
for (i in 7:491){
  a <- table(Ready[,i])
  b <- a["2"]/sum(a)
  valores <- c(valores, b)
}

burden_table$Burden <- valores

race <- read.delim("/home/roberto/Sorghum/Evol_model/Burden/race", header = F) %>%
  `colnames<-`(c("ID", "Race"))

only_race <- race %>%
  inner_join(burden_table, by="ID")


pal <- wes_palette("FantasticFox1", 8, type = "continuous")
wes <- wes_palette("Cavalcanti1")
chev <- wes_palette("Chevalier1")
fox <- wes_palette("FantasticFox1")

colors <- c( fox[1], fox[3], "grey", wes[1], chev[1],"red", "gray31", fox[5])
colors <- c( "darkblue", "grey", "grey", "grey", "grey", "grey", fox[5], fox[5])

p1<-ggplot(only_race, aes(x=Race, y=Burden, fill=Race)) +
  geom_boxplot() +
  scale_fill_manual(values = colors)+
  theme(axis.line = element_line(size=1, colour = "black"),
        panel.grid.major = element_line(colour = "#d3d3d3"), panel.grid.minor = element_blank(),
        panel.border = element_blank(), panel.background = element_blank()) +
  #scale_y_continuous(labels = dollar_format(suffix = "%", prefix = "")) +
  theme(plot.title = element_text(size = 14, face = "bold"),
        axis.text.x=element_text(colour="black", size = 10),
        axis.text.y=element_text(colour="black", size = 10),
        legend.key = element_rect(fill = "transparent", colour = "transparent")) +
  theme(legend.position="none", legend.direction="vertical",
        legend.title = element_blank()) +
  ylim(0,0.125)

p1

ggsave("/home/roberto/Sorghum/Figures/burden_all.png", plot= p1, device = "png", scale = 1, width = 9, height = 5, units = c("in"),dpi = 750, limitsize = TRUE)


```

## Only Conserved DEL

```{r}

Ready <- Ready %>%
  filter(SET == "Deleterious")

header <- read.delim("/home/roberto/Sorghum/Evol_model/Burden/header") 

burden_table <- as.data.frame(matrix(nrow= 485,ncol=2))
burden_table[,1] <- header
colnames(burden_table) <- c("ID", "Burden")


valores <- vector()
for (i in 7:491){
  a <- table(Ready[,i])
  b <- a["2"]/sum(a)
  valores <- c(valores, b)
}

burden_table$Burden <- valores

race <- read.delim("/home/roberto/Sorghum/Evol_model/Burden/race", header = F) %>%
  `colnames<-`(c("ID", "Race"))

only_race <- race %>%
  inner_join(burden_table, by="ID")


pal <- wes_palette("FantasticFox1", 8, type = "continuous")
wes <- wes_palette("Cavalcanti1")
chev <- wes_palette("Chevalier1")
fox <- wes_palette("FantasticFox1")

colors <- c( fox[1], fox[3], "grey", wes[1], chev[1],"red", "gray31", fox[5])
colors <- c( "darkblue", "grey", "grey", "grey", "grey", "grey", fox[5], fox[5])

p<-ggplot(only_race, aes(x=Race, y=Burden, fill=Race)) +
  geom_boxplot() +
  scale_fill_manual(values = colors)+
  theme(axis.line = element_line(size=1, colour = "black"),
        panel.grid.major = element_line(colour = "#d3d3d3"), panel.grid.minor = element_blank(),
        panel.border = element_blank(), panel.background = element_blank()) +
  #scale_y_continuous(labels = dollar_format(suffix = "%", prefix = "")) +
  theme(plot.title = element_text(size = 14, face = "bold"),
        axis.text.x=element_text(colour="black", size = 10),
        axis.text.y=element_text(colour="black", size = 10),
        legend.key = element_rect(fill = "transparent", colour = "transparent")) +
  theme(legend.position="none", legend.direction="vertical",
        legend.title = element_blank()) +
  ylim(0,0.125)
  
p


ggsave("/home/roberto/Sorghum/Figures/burden_del.png", plot= p, device = "png", scale = 1, width = 9, height = 5, units = c("in"),dpi = 750, limitsize = TRUE)

```

## Printing

```{r, fig.width=14, fig.height=4}

p1 + p


```

## MAF > 0.05

```{r,fig.width=6, fig.height=3}

#Calculating different burden levels

Ready_del <- Master %>%
  filter(MAF > 0.05)

header <- read.delim("/home/roberto/Sorghum/Evol_model/Burden/header") 

burden_table <- as.data.frame(matrix(nrow= 485,ncol=2))
burden_table[,1] <- header
colnames(burden_table) <- c("ID", "Burden")


valores <- vector()
for (i in 7:491){
  a <- table(Ready_del[,i])
  b <- a["2"]/sum(a)
  valores <- c(valores, b)
}

burden_table$Burden <- valores

race <- read.delim("/home/roberto/Sorghum/Evol_model/Burden/race", header = F) %>%
  `colnames<-`(c("ID", "Race"))

only_race <- race %>%
  inner_join(burden_table, by="ID")

pal <- wes_palette("FantasticFox1", 8, type = "continuous")
wes <- wes_palette("Cavalcanti1")
chev <- wes_palette("Chevalier1")
fox <- wes_palette("FantasticFox1")

colors <- c( fox[1], fox[3], "grey", wes[1], chev[1],"red", "gray31", fox[5])

p<-ggplot(only_race, aes(x=Race, y=Burden, fill=Race)) +
  geom_boxplot() +
  scale_fill_manual(values = colors)+
  theme(axis.line = element_line(size=1, colour = "black"),
        panel.grid.major = element_line(colour = "#d3d3d3"), panel.grid.minor = element_blank(),
        panel.border = element_blank(), panel.background = element_blank()) +
  #scale_y_continuous(labels = dollar_format(suffix = "%", prefix = "")) +
  theme(plot.title = element_text(size = 14, face = "bold"),
        axis.text.x=element_text(colour="black", size = 10),
        axis.text.y=element_text(colour="black", size = 10),
        legend.key = element_rect(fill = "transparent", colour = "transparent")) 
p


```

# BURDEN LANDRACES/CULTIVATED/WW

```{r, fig.height=5, fig.width=3.5}


library(ggplot2)
library(scales)
library(wesanderson)
#Calculating different burden levels


Master <- Ready
Ready_del <- Master 

header <- read.delim("/home/roberto/Sorghum/Evol_model/Burden/header") 

burden_table <- as.data.frame(matrix(nrow= 485,ncol=2))
burden_table[,1] <- header
colnames(burden_table) <- c("ID", "Burden")


valores <- vector()
for (i in 7:491){
  a <- table(Ready_del[,i])
  b <- a["2"]/sum(a)
  valores <- c(valores, b)
}


burden_table$Burden <- valores

race <- read.delim("/home/roberto/Sorghum/Evol_model/Burden/race_version2", header = F) %>%
  `colnames<-`(c("ID", "Race"))

only_race <- race %>%
  inner_join(burden_table, by="ID") %>%
  mutate(Race= recode(Race, "Improved varieties" = "Improved")) %>%
  mutate(Race= recode(Race, "Wild and weedy" = "Wild")) %>%
  mutate(Race= recode(Race, "Landraces" = "Landrace"))
AllTAB <- only_race

wes <- wes_palette("Cavalcanti1")
wesi <- c(wes[1], wes[2], wes[5])

Sorghum_all<-ggplot(only_race, aes(x=Race, y=Burden, fill=Race)) +
  geom_boxplot() +
  scale_fill_manual(values = wesi)+
  theme(axis.line = element_line(size=1, colour = "black"),
        panel.grid.major = element_line(colour = "#d3d3d3"), panel.grid.minor = element_blank(),
        panel.border = element_blank(), panel.background = element_blank()) +
  #scale_y_continuous(labels = dollar_format(suffix = "%", prefix = "")) +
  theme(plot.title = element_text(size = 14, face = "bold"),
        axis.text.x=element_text(colour="black", size = 10),
        axis.text.y=element_text(colour="black", size = 10),
        legend.key = element_rect(fill = "transparent", colour = "transparent")) +
  theme(legend.position="none", legend.direction="vertical",
        legend.title = element_blank())

Sorghum_all



```

# Private alleles
```{r}

## WILDS

wilds <- only_race %>%
  filter(Race == "Wild")
wildID <- wilds$ID
keep <- c(colnames(Master)[1:6], wildID)

wilddel <- Master %>%
  select(keep) %>%
  mutate(sumVar = rowSums(.[7:13], na.rm=T)) %>%
  filter(sumVar > 1)

wildSNP <- wilddel$CHR_POS
  
##LANDRACES

land <- only_race %>%
  filter(Race == "Landrace")
landID <- land$ID
keep <- c(colnames(Master)[1:6], landID)

landdel <- Master %>%
  select(keep) %>%
  mutate(sumVar = rowSums(.[7:23], na.rm=T)) %>%
  filter(sumVar > 1)

landSNP <- landdel$CHR_POS

##IMPROVED

impro <- only_race %>%
  filter(Race == "Improved")
improID <- impro$ID
keep <- c(colnames(Master)[1:6], improID)

improdel <- Master %>%
  select(keep) %>%
  mutate(sumVar = rowSums(.[7:19], na.rm=T)) %>%
  filter(sumVar > 1)

improSNP <- improdel$CHR_POS

inter <- Reduce(intersect, list(improSNP,landSNP,wildSNP))

```


## Burden on SEG variants

```{r}
library(ggplot2)
library(scales)
library(wesanderson)
#Calculating different burden levels

Ready_del <- Master %>%
  filter(CHR_POS %in% inter)

header <- read.delim("/home/roberto/Sorghum/Evol_model/Burden/header") 

burden_table <- as.data.frame(matrix(nrow= 485,ncol=2))
burden_table[,1] <- header
colnames(burden_table) <- c("ID", "Burden")


valores <- vector()
for (i in 7:491){
  a <- table(Ready_del[,i])
  b <- a["2"]/sum(a)
  valores <- c(valores, b)
}


burden_table$Burden <- valores

race <- read.delim("/home/roberto/Sorghum/Evol_model/Burden/race_version2", header = F) %>%
  `colnames<-`(c("ID", "Race"))

only_race <- race %>%
  inner_join(burden_table, by="ID") %>%
  mutate(Race= recode(Race, "Improved varieties" = "Improved")) %>%
  mutate(Race= recode(Race, "Wild and weedy" = "Wild")) %>%
  mutate(Race= recode(Race, "Landraces" = "Landrace"))
segTAB <- only_race

wes <- wes_palette("Cavalcanti1")
wesi <- c(wes[1], wes[2], wes[5])

Sorghum_seg<-ggplot(only_race, aes(x=Race, y=Burden, fill=Race)) +
  geom_boxplot() +
  scale_fill_manual(values = wesi)+
  theme(axis.line = element_line(size=1, colour = "black"),
        panel.grid.major = element_line(colour = "#d3d3d3"), panel.grid.minor = element_blank(),
        panel.border = element_blank(), panel.background = element_blank()) +
  #scale_y_continuous(labels = dollar_format(suffix = "%", prefix = "")) +
  theme(plot.title = element_text(size = 14, face = "bold"),
        axis.text.x=element_text(colour="black", size = 10),
        axis.text.y=element_text(colour="black", size = 10),
        legend.key = element_rect(fill = "transparent", colour = "transparent")) +
  theme(legend.position="none", legend.direction="vertical",
        legend.title = element_blank()) 

Sorghum_seg

```

## Burden FIXED variants

```{r}


#########################
## WILDS

#wilds <- only_race %>%
#  filter(Race == "Wild")
#wildID <- wilds$ID
#keep <- c(colnames(Master)[1:6], wildID)

#wilddel <- Master %>%
#  select(keep) %>%
#  mutate(sumVar = rowSums(.[7:13], na.rm=T)) 

#table(wilddel$sumVar)%>%
#  filter(sumVar == 0 | sumVar == 14)

#wildSNP <- wilddel$CHR_POS

##LANDRACES

land <- only_race %>%
  filter(Race == "Landrace")
landID <- land$ID
keep <- c(colnames(Master)[1:6], landID)


landdel <- Master %>%
  select(keep) %>%
  mutate(sumVar = rowSums(.[7:23], na.rm=T)) %>%
  filter(sumVar == 0 | sumVar == 34)


landSNP <- landdel$CHR_POS

##IMPROVED

impro <- only_race %>%
  filter(Race == "Improved")
improID <- impro$ID
keep <- c(colnames(Master)[1:6], improID)

improdel <- Master %>%
  select(keep) %>%
  mutate(sumVar = rowSums(.[7:19], na.rm=T)) %>%
  filter(sumVar == 0 | sumVar == 40)


improSNP <- improdel$CHR_POS

inter <- Reduce(union, list(improSNP,landSNP,wildSNP))

Ready_del <- Master %>%
  filter(CHR_POS %in% inter)

header <- read.delim("/home/roberto/Sorghum/Evol_model/Burden/header") 

burden_table <- as.data.frame(matrix(nrow= 485,ncol=2))
burden_table[,1] <- header
colnames(burden_table) <- c("ID", "Burden")


valores <- vector()
for (i in 7:491){
  a <- table(Ready_del[,i])
  b <- a["2"]/sum(a)
  valores <- c(valores, b)
}


burden_table$Burden <- valores

race <- read.delim("/home/roberto/Sorghum/Evol_model/Burden/race_version2", header = F) %>%
  `colnames<-`(c("ID", "Race"))

only_race <- race %>%
  inner_join(burden_table, by="ID") %>%
  mutate(Race= recode(Race, "Improved varieties" = "Improved")) %>%
  mutate(Race= recode(Race, "Wild and weedy" = "Wild")) %>%
  mutate(Race= recode(Race, "Landraces" = "Landrace"))
fixTAB <- only_race

wes <- wes_palette("Cavalcanti1")
wesi <- c(wes[1], wes[2], wes[5])

Sorghum_fix<-ggplot(only_race, aes(x=Race, y=Burden, fill=Race)) +
  geom_boxplot() +
  scale_fill_manual(values = wesi)+
  theme(axis.line = element_line(size=1, colour = "black"),
        panel.grid.major = element_line(colour = "#d3d3d3"), panel.grid.minor = element_blank(),
        panel.border = element_blank(), panel.background = element_blank()) +
  #scale_y_continuous(labels = dollar_format(suffix = "%", prefix = "")) +
  theme(plot.title = element_text(size = 14, face = "bold"),
        axis.text.x=element_text(colour="black", size = 10),
        axis.text.y=element_text(colour="black", size = 10),
        legend.key = element_rect(fill = "transparent", colour = "transparent")) +
  theme(legend.position="none", legend.direction="vertical",
        legend.title = element_blank()) 

Sorghum_fix



```



## Burden on del PRIVATE variants # (skip)

```{r}

#WILD privates

together <- c(landSNP, improSNP)

wildprivate <- wilddel %>%
  filter(!CHR_POS %in% together)

burden_table <- as.data.frame(matrix(nrow= 7,ncol=2))
burden_table[,1] <- wildID
colnames(burden_table) <- c("ID", "Burden")

valores <- vector()
for (i in 7:13){
  a <- table(wildprivate[,i])
  b <- a["2"]/sum(a)
  valores <- c(valores, b)
}


wildTAB <- burden_table %>%
  mutate(Race = "Wild")
wildTAB$Burden <- valores

#LAND privates

together <- c(wildSNP, improSNP)

landprivate <- landdel %>%
  filter(!CHR_POS %in% together)

burden_table <- as.data.frame(matrix(nrow= 17,ncol=2))
burden_table[,1] <- landID
colnames(burden_table) <- c("ID", "Burden")

valores <- vector()
for (i in 7:23){
  a <- table(landprivate[,i])
  b <- a["2"]/sum(a)
  valores <- c(valores, b)
}

landTAB <- burden_table %>%
  mutate(Race = "Landrace")
landTAB$Burden <- valores

#Impro privates

together <- c(landSNP, wildSNP)

improprivate <- improdel %>%
  filter(!CHR_POS %in% together)

burden_table <- as.data.frame(matrix(nrow= 13,ncol=2))
burden_table[,1] <- improID
colnames(burden_table) <- c("ID", "Burden")

valores <- vector()
for (i in 7:19){
  a <- table(improprivate[,i])
  b <- a["2"]/sum(a)
  valores <- c(valores, b)
}

improTAB <- burden_table %>%
  mutate(Race = "Improved")
improTAB$Burden <- valores

only_race <- rbind(wildTAB, landTAB, improTAB) 




## PLOT

Privates <-ggplot(only_race, aes(x=Race, y=Burden, fill=Race)) +
  geom_boxplot() +
  scale_fill_manual(values = wesi)+
  theme(axis.line = element_line(size=1, colour = "black"),
        panel.grid.major = element_line(colour = "#d3d3d3"), panel.grid.minor = element_blank(),
        panel.border = element_blank(), panel.background = element_blank()) +
  #scale_y_continuous(labels = dollar_format(suffix = "%", prefix = "")) +
  theme(plot.title = element_text(size = 14, face = "bold"),
        axis.text.x=element_text(colour="black", size = 10),
        axis.text.y=element_text(colour="black", size = 10),
        legend.key = element_rect(fill = "transparent", colour = "transparent")) +
  ylab("")

Privates

```

## Plotting Sorghum

```{r}


segTAB <- segTAB %>%
  mutate(TYPE = "Seg")

AllTAB <- AllTAB %>%
  mutate(TYPE = "All")

fixTAB <- fixTAB %>%
  mutate(TYPE = "Fix")

wes <- wes_palette("Cavalcanti1")
wesi <- c(wes[1], wes[4], wes[5])

toplot <- rbind(segTAB, AllTAB, fixTAB)
Sorghum_final <-ggplot(toplot, aes(x=TYPE, y=Burden, fill=Race)) +
  geom_boxplot(lwd=0.2, outlier.size = 0.4) +
  scale_fill_manual(values = wesi)+
  theme(axis.line = element_line(size=1, colour = "black"),
        panel.grid.major = element_line(colour = "#d3d3d3"), panel.grid.minor = element_blank(),
        panel.border = element_blank(), panel.background = element_blank()) +
  #scale_y_continuous(labels = dollar_format(suffix = "%", prefix = "")) +
  theme(plot.title = element_text(size = 14, face = "bold"),
        axis.text.x=element_text(colour="black", size = 10),
        axis.text.y=element_text(colour="black", size = 10),
        legend.key = element_rect(fill = "transparent", colour = "transparent")) +
  theme(legend.position="none", legend.direction="vertical",
        legend.title = element_blank()) +
  ylab("Burden")

Sorghum_final


```
```{r, fig.width=3.5, fig.height=5}


segTAB <- segTAB %>%
  mutate(TYPE = "Seg")

AllTAB <- AllTAB %>%
  mutate(TYPE = "All")

fixTAB <- fixTAB %>%
  mutate(TYPE = "Fix")

wes <- wes_palette("Cavalcanti1")
wesi <- c(wes[1], wes[4], wes[5])

toplot <- rbind(segTAB, fixTAB)
Sorghum_final <-ggplot(toplot, aes(x=TYPE, y=Burden, fill=Race)) +
  geom_boxplot(lwd=0.2, outlier.size = 0.4) +
  scale_fill_manual(values = wesi)+
  theme(axis.line = element_line(size=1, colour = "black"),
        panel.grid.major = element_line(colour = "#d3d3d3"), panel.grid.minor = element_blank(),
        panel.border = element_blank(), panel.background = element_blank()) +
  #scale_y_continuous(labels = dollar_format(suffix = "%", prefix = "")) +
  theme(plot.title = element_text(size = 14, face = "bold"),
        axis.text.x=element_text(colour="black", size = 10),
        axis.text.y=element_text(colour="black", size = 10),
        legend.key = element_rect(fill = "transparent", colour = "transparent")) +
  theme(legend.position="none", legend.direction="vertical",
        legend.title = element_blank()) +
  ylab("Burden")

Sorghum_final


```


# BURDEN MAIZE

## Select Ancestral alleles

```{r}

header <- read.delim("/home/roberto/Maize/Comparative_Hapmap/Burden/true_header", header = F, stringsAsFactors = F) 

dosage <- read.delim("/home/roberto/Maize/Comparative_Hapmap/Burden/Burden_for_R", header = T) %>%
  `colnames<-`(header[,1])
dosage <- dosage[,-72]

daf <- read.delim(file = "/home/roberto/Sorghum/Evol_model/Derived_alleles/MAIZE/READY_to_run/splitting/DAFtoR", header = F) %>% 
  `colnames<-`(c("link","DAF_p"))

dosage <- dosage %>%
  unite(link, CHR, POS) %>%
  right_join(daf, by="link")%>%
  mutate(Class = case_when(DAF_p >=0.55 ~ "Ancestral",
                            DAF_p <=0.45 ~ "Derived"))


# Modify encoding : Ancestral allele = 0, Derived allele = 1

derivao <- dosage %>%
  filter(Class == "Derived")

ancestral <- dosage %>%
  filter(Class == "Ancestral") 

for (i in 4:96){
  derivao[,i][derivao[,i] == 0] <- "ZERO"
  derivao[,i][derivao[,i] == 2] <- "TWO"
}

for (i in 4:96){
  derivao[,i][derivao[,i] == "ZERO"] <- as.numeric(2)
  derivao[,i][derivao[,i] == "TWO"] <- as.numeric(0)
  derivao[,i] <- as.numeric(derivao[,i])
}
  

##
Ready <- rbind(ancestral, derivao)

```



```{r,fig.height=5, fig.width=3.5}

#header <- read.delim("/home/roberto/Maize/Comparative_Hapmap/Burden/true_header", header = F, stringsAsFactors = F) 

#dosage <- read.delim("/home/roberto/Maize/Comparative_Hapmap/Burden/Burden_for_R", header = T) %>%
#  `colnames<-`(header[,1])

#dosage <- dosage[,-40]
#header <- header[-40,1]

dosage <- Ready

header <- colnames(dosage)

burden_table <- as.data.frame(matrix(nrow= 93,ncol=2))
burden_table[,1] <- header[4:96]
colnames(burden_table) <- c("ID", "Burden")

valores <- vector()
for (i in 4:96){
  a <- table(dosage[,i])
  b <- a["2"]/sum(a)
  valores <- c(valores, b)
}

burden_table$Burden <- valores

race <- read.delim("/home/roberto/Maize/Comparative_Hapmap/Burden/Races", header = F) %>%
  `colnames<-`(c("ID", "Race"))

only_race <- race %>%
  left_join(burden_table, by="ID") %>%
  mutate(Race= recode(Race, "wild" = "Wild"))
AllTAB <- only_race


wes <- wes_palette("Cavalcanti1")
wesi <- c(wes[1], wes[2], wes[5])

Maize_burden<-ggplot(only_race, aes(x=Race, y=Burden, fill=Race)) +
  geom_boxplot() +
  scale_fill_manual(values = wesi)+
  theme(axis.line = element_line(size=1, colour = "black"),
        panel.grid.major = element_line(colour = "#d3d3d3"), panel.grid.minor = element_blank(),
        panel.border = element_blank(), panel.background = element_blank()) +
  #scale_y_continuous(labels = dollar_format(suffix = "%", prefix = "")) +
  theme(plot.title = element_text(size = 14, face = "bold"),
        axis.text.x=element_text(colour="black", size = 10),
        axis.text.y=element_text(colour="black", size = 10),
        legend.key = element_rect(fill = "transparent", colour = "transparent")) +
  theme(legend.position="none", legend.direction="vertical",
        legend.title = element_blank()) +
  ylab("")

Maize_burden


```


## Segregating Maize

```{r}

## WILDS

Ready #<- Ready %>%
  #separate(col = link, into = c("CHR", "POS"), sep = "_")

wilds <- only_race %>%
  filter(Race == "Wild")
wildID <- wilds$ID
keep <- c(colnames(Ready)[1:3], wildID)

wilddel <- Ready %>%
  select(keep) %>%
  mutate(sumVar = rowSums(.[4:23], na.rm=T)) %>%
  filter(sumVar > 1)

wildSNP <- wilddel$link
  
## LANDRACES

land <- only_race %>%
  filter(Race == "Landrace")
landID <- land$ID
keep <- c(colnames(Ready)[1:3], landID)

landdel <- dosage %>%
  select(keep) %>%
  mutate(sumVar = rowSums(.[4:28], na.rm=T)) %>%
  filter(sumVar > 1)

landSNP <- landdel$link

## IMPROVED

impro <- only_race %>%
  filter(Race == " Improved")
improID <- impro$ID
keep <- c(colnames(Ready)[1:3], improID)

improdel <- Ready %>%
  select(keep) %>%
  mutate(sumVar = rowSums(.[4:52], na.rm=T)) %>%
  filter(sumVar > 1)

improSNP <- improdel$link

inter <- Reduce(intersect, list(improSNP,landSNP,wildSNP))

##################

seg <- Ready %>%
  filter(link %in% inter)
  
burden_table <- as.data.frame(matrix(nrow= 94,ncol=2))
burden_table[,1] <- header[5:98]
colnames(burden_table) <- c("ID", "Burden")

valores <- vector()
for (i in 5:98){
  a <- table(seg[,i])
  b <- a["2"]/sum(a)
  valores <- c(valores, b)
}

burden_table$Burden <- valores

race <- read.delim("/home/roberto/Maize/Comparative_Hapmap/Burden/Races", header = F) %>%
  `colnames<-`(c("ID", "Race"))

only_race <- race %>%
  left_join(burden_table, by="ID") %>%
  mutate(Race= recode(Race, "wild" = "Wild"))
segTAB <- only_race

wes <- wes_palette("Cavalcanti1")
wesi <- c(wes[1], wes[2], wes[5])

Maize_seg<-ggplot(only_race, aes(x=Race, y=Burden, fill=Race)) +
  geom_boxplot() +
  scale_fill_manual(values = wesi)+
  theme(axis.line = element_line(size=1, colour = "black"),
        panel.grid.major = element_line(colour = "#d3d3d3"), panel.grid.minor = element_blank(),
        panel.border = element_blank(), panel.background = element_blank()) +
  theme(plot.title = element_text(size = 14, face = "bold"),
        axis.text.x=element_text(colour="black", size = 10),
        axis.text.y=element_text(colour="black", size = 10),
        legend.key = element_rect(fill = "transparent", colour = "transparent")) +
  ylab("")

Maize_seg


```

## NON segregating

```{r}

## WILDS

Ready #<- Ready %>%
  #separate(col = link, into = c("CHR", "POS"), sep = "_")

wilds <- only_race %>%
  filter(Race == "Wild")
wildID <- wilds$ID
keep <- c(colnames(Ready)[1:3], wildID)

wilddel <- Ready %>%
  select(keep) %>%
  mutate(sumVar = rowSums(.[4:23], na.rm=T)) %>%
  filter(sumVar > 1)

wildSNP <- wilddel$link
  
## LANDRACES

land <- only_race %>%
  filter(Race == "Landrace")
landID <- land$ID
keep <- c(colnames(Ready)[1:3], landID)

landdel <- dosage %>%
  select(keep) %>%
  mutate(sumVar = rowSums(.[4:28], na.rm=T)) %>%
  filter(sumVar > 1)

landSNP <- landdel$link

## IMPROVED

impro <- only_race %>%
  filter(Race == " Improved")
improID <- impro$ID
keep <- c(colnames(Ready)[1:3], improID)

improdel <- Ready %>%
  select(keep) %>%
  mutate(sumVar = rowSums(.[4:52], na.rm=T)) %>%
  filter(sumVar > 1)

improSNP <- improdel$link

inter <- Reduce(intersect, list(improSNP,landSNP,wildSNP))

##################

seg <- Ready %>%
  filter(!link %in% inter)
  
burden_table <- as.data.frame(matrix(nrow= 94,ncol=2))
burden_table[,1] <- header[5:98]
colnames(burden_table) <- c("ID", "Burden")

valores <- vector()
for (i in 5:98){
  a <- table(seg[,i])
  b <- a["2"]/sum(a)
  valores <- c(valores, b)
}

burden_table$Burden <- valores

race <- read.delim("/home/roberto/Maize/Comparative_Hapmap/Burden/Races", header = F) %>%
  `colnames<-`(c("ID", "Race"))

only_race <- race %>%
  left_join(burden_table, by="ID") %>%
  mutate(Race= recode(Race, "wild" = "Wild"))
fixTAB <- only_race

wes <- wes_palette("Cavalcanti1")
wesi <- c(wes[1], wes[2], wes[5])

Maize_fix<-ggplot(only_race, aes(x=Race, y=Burden, fill=Race)) +
  geom_boxplot() +
  scale_fill_manual(values = wesi)+
  theme(axis.line = element_line(size=1, colour = "black"),
        panel.grid.major = element_line(colour = "#d3d3d3"), panel.grid.minor = element_blank(),
        panel.border = element_blank(), panel.background = element_blank()) +
  theme(plot.title = element_text(size = 14, face = "bold"),
        axis.text.x=element_text(colour="black", size = 10),
        axis.text.y=element_text(colour="black", size = 10),
        legend.key = element_rect(fill = "transparent", colour = "transparent")) +
  ylab("")

Maize_fix


```



## FIXED Maize ## alternative way, do not count

 
```{r}

## WILD

wilds <- only_race %>%
  filter(Race == "Wild")
wildID <- wilds$ID
keep <- c(colnames(dosage)[1:4], wildID)

wilddel <- dosage %>%
  select(keep) %>%
  mutate(sumVar = rowSums(.[5:24], na.rm=T)) %>%
  filter(sumVar == 0 | sumVar == 40)

wildSNP <- wilddel$link

## LANDRACES

land <- only_race %>%
  filter(Race == "Landrace")
landID <- land$ID
keep <- c(colnames(dosage)[1:4], landID)

landdel <- dosage %>%
  select(keep) %>%
  mutate(sumVar = rowSums(.[5:29], na.rm=T)) %>%
  filter(sumVar == 0 | sumVar == 38)

landSNP <- landdel$link

## IMPROVED

impro <- only_race %>%
  filter(Race == " Improved")
improID <- impro$ID
keep <- c(colnames(dosage)[1:4], improID)

improdel <- dosage %>%
  select(keep) %>%
  mutate(sumVar = rowSums(.[5:53], na.rm=T)) %>%
  filter(sumVar == 0 | sumVar == 70)

improSNP <- paste(improdel$CHR, improdel$POS, sep = "_")


inter <- Reduce(union, list(improSNP,landSNP,wildSNP))

##################

fixed <- dosage %>%
  mutate(CHR_POS = paste(CHR, POS, sep = "_"))%>%
  filter(CHR_POS %in% inter)
  

burden_table <- as.data.frame(matrix(nrow= 94,ncol=2))
burden_table[,1] <- header[5:98]
colnames(burden_table) <- c("ID", "Burden")


valores <- vector()
for (i in 5:98){
  a <- table(fixed[,i])
  b <- a["2"]/sum(a)
  valores <- c(valores, b)
}

burden_table$Burden <- valores

race <- read.delim("/home/roberto/Maize/Comparative_Hapmap/Burden/Races", header = F) %>%
  `colnames<-`(c("ID", "Race"))

only_race <- race %>%
  left_join(burden_table, by="ID") %>%
  mutate(Race= recode(Race, "wild" = "Wild"))
fixTAB <- only_race

wes <- wes_palette("Cavalcanti1")
wesi <- c(wes[1], wes[2], wes[5])

Maize_burden<-ggplot(only_race, aes(x=Race, y=Burden, fill=Race)) +
  geom_boxplot() +
  scale_fill_manual(values = wesi)+
  theme(axis.line = element_line(size=1, colour = "black"),
        panel.grid.major = element_line(colour = "#d3d3d3"), panel.grid.minor = element_blank(),
        panel.border = element_blank(), panel.background = element_blank()) +
  #scale_y_continuous(labels = dollar_format(suffix = "%", prefix = "")) +
  theme(plot.title = element_text(size = 14, face = "bold"),
        axis.text.x=element_text(colour="black", size = 10),
        axis.text.y=element_text(colour="black", size = 10),
        legend.key = element_rect(fill = "transparent", colour = "transparent")) +
  ylab("")

Maize_burden


```

## Plotting Maize

```{r}


segTAB <- segTAB %>%
  mutate(TYPE = "Seg")

AllTAB <- AllTAB %>%
  mutate(TYPE = "All")

fixTAB <- fixTAB %>%
  mutate(TYPE = "Fix")

wes <- wes_palette("Cavalcanti1")
wesi <- c(wes[1], wes[4], wes[5])

toplot <- rbind(segTAB, AllTAB, fixTAB)
Maize_final <-ggplot(toplot, aes(x=TYPE, y=Burden, fill=Race)) +
  geom_boxplot(lwd=0.2, outlier.size = 0.4) +
  scale_fill_manual(values = wesi)+
  theme(axis.line = element_line(size=1, colour = "black"),
        panel.grid.major = element_line(colour = "#d3d3d3"), panel.grid.minor = element_blank(),
        panel.border = element_blank(), panel.background = element_blank()) +
  #scale_y_continuous(labels = dollar_format(suffix = "%", prefix = "")) +
  theme(plot.title = element_text(size = 14, face = "bold"),
        axis.text.x=element_text(colour="black", size = 10),
        axis.text.y=element_text(colour="black", size = 10),
        legend.key = element_rect(fill = "transparent", colour = "transparent")) +
  ylab("")

Maize_final


```


```{r,fig.width=3.5, fig.height=5 }

segTAB <- segTAB %>%
  mutate(TYPE = "Seg")

fixTAB <- fixTAB %>%
  mutate(TYPE = "Fix")

wes <- wes_palette("Cavalcanti1")
wesi <- c(wes[1], wes[4], wes[5])

toplot <- rbind(segTAB, fixTAB)
Maize_final <-ggplot(toplot, aes(x=TYPE, y=Burden, fill=Race)) +
  geom_boxplot(lwd=0.2, outlier.size = 0.4) +
  scale_fill_manual(values = wesi)+
  theme(axis.line = element_line(size=1, colour = "black"),
        panel.grid.major = element_line(colour = "#d3d3d3"), panel.grid.minor = element_blank(),
        panel.border = element_blank(), panel.background = element_blank()) +
  #scale_y_continuous(labels = dollar_format(suffix = "%", prefix = "")) +
  theme(plot.title = element_text(size = 14, face = "bold"),
        axis.text.x=element_text(colour="black", size = 10),
        axis.text.y=element_text(colour="black", size = 10),
        legend.key = element_rect(fill = "transparent", colour = "transparent")) +
  theme(legend.position="none", legend.direction="vertical",
        legend.title = element_blank()) +
  ylab("")

Maize_final



```




# FIGURE 2

```{r, fig.width= 8.5, fig.height=7}
library(patchwork)

Merged <- Graph + (Sorghum_del + Maize_del) + (Sorghum_final + Maize_final) + plot_layout(ncol = 1)  
Merged

ggsave("/home/roberto/Sorghum/Figures/Fig2_full_ultimate.png", plot= Merged, device = "png", scale = 1, width = 9, height = 6, units = c("in"),dpi = 750, limitsize = TRUE)

```

Add a new chunk by clicking the *Insert Chunk* button on the toolbar or by pressing *Ctrl+Alt+I*.

When you save the notebook, an HTML file containing the code and output will be saved alongside it (click the *Preview* button or press *Ctrl+Shift+K* to preview the HTML file).

The preview shows you a rendered HTML copy of the contents of the editor. Consequently, unlike *Knit*, *Preview* does not run any R code chunks. Instead, the output of the chunk when it was last run in the editor is displayed.
